testsuite/src/it/java/io/quarkus/ts/startstop/SpecialCharsTest.java

```diff
@@ -0,0 +1,241 @@
+        try {
+            // Clean target directory
+            cleanTarget(app);
+
+            // Delete dir with special chars
+            if (appDestDir.exists()) {
+                FileUtils.deleteDirectory(appDestDir);
+            }
+
+            // Make dir with special chars
+            if (!appDestDir.mkdir()) {
+                throw new RuntimeException("Cannot create directory " + appDestDir);
+            }
+
+            // Copy to path with special characters
+            FileUtils.copyDirectoryToDirectory(appBaseDir, appDestDir);
+
+            // Replace relative path to parent project
+            Path path = appPomXml.toPath();
+            String content = new String(Files.readAllBytes(path), StandardCharsets.UTF_8);
+            content = content.replaceAll("<relativePath>..</relativePath>", "<relativePath>../..</relativePath>");
+            Files.write(path, content.getBytes(StandardCharsets.UTF_8));
+
+            // Create logs directory
+            Files.createDirectories(logsDir.toPath());
+
+            List<String> cmd;
+            // Build
+            if (mvnCmds != MvnCmds.DEV) {
+                buildLogA = new File(logsDir + File.separator + subdir + "-" + mvnCmds.name().toLowerCase() + "-build.log");
+                ExecutorService buildService = Executors.newFixedThreadPool(1);
+                List<String> baseBuildCmd = new ArrayList<>();
+                baseBuildCmd.addAll(Arrays.asList(mvnCmds.mvnCmds[0]));
+                baseBuildCmd.add("-Dquarkus.version=" + getQuarkusVersion());
+                cmd = getBuildCommand(baseBuildCmd.toArray(new String[0]));
+
+                appendln(whatIDidReport, "# " + cn + ", " + mn);
+                appendln(whatIDidReport, (new Date()).toString());
+                appendln(whatIDidReport, appDir.getAbsolutePath());
+                appendlnSection(whatIDidReport, String.join(" ", cmd));
+
+                LOGGER.info("Building (" + cmd + ")");
+                buildService.submit(new Commands.ProcessRunner(appDir, buildLogA, cmd, 20));
+
+                buildService.shutdown();
+                buildService.awaitTermination(30, TimeUnit.MINUTES);
+
+                assertTrue(buildLogA.exists());
+                checkLog(cn, mn, app, mvnCmds, buildLogA);
+            }
+
+            // Run
+            runLogA = new File(logsDir + File.separator + subdir +  "-" + mvnCmds.name().toLowerCase() + "-run.log");
+            cmd = getRunCommand(mvnCmds.mvnCmds[mvnCmds != MvnCmds.DEV ? 1 : 0]);
+            LOGGER.info("Running (" + cmd + ")");
+            pA = runCommand(cmd, appDir, runLogA);
+
+            appendln(whatIDidReport, appDir.getAbsolutePath());
+            appendlnSection(whatIDidReport, String.join(" ", cmd));
+
+            // Test web page
+            LOGGER.info("Testing web page content...");
+            int timeout = mvnCmds != MvnCmds.DEV ? 5 : 20;
```

Comment:
I received timeout after 20s when running `mvn clean verify -Ptestsuite -DincludeTags=special-chars -DexcludeTags=native -Dquarkus.version=1.3.4.Final-redhat-00001  -Dquarkus.platform.version=1.3.4.Final-redhat-00001 -Dtests.maven.repo.local=/Users/rsvoboda/Downloads/maven-repository  -Dmaven.repo.local=/Users/rsvoboda/Downloads/maven-repository`

This was because Maven was still downloading dependencies.

I see 2 options here:
 - increase the timeout to higher number - e.g. 60 seconds
 - implement warmup logic as in `ArtifactGeneratorTest`